### ISSUE TITLE
ConfigWatch removing a property from Consul is ignored

### ISSUE THREAD

**Issue by ShalomDeitch1** [spring-cloud/spring-cloud-config#636]
version: spring-cloud-consul-config-1.2.0.M1
package: org.springframework.cloud.consul.config
class: ConfigWatch

line 98-100
					// if response.value == null, response was a 404, otherwise it was a 200
					// reducing churn if there wasn't anything
					if (response.getValue() != null && !response.getValue().isEmpty()) {

If there was a value and it was removed, the response.getValue() with be null.
However, removing a value is important information, e.g. when there are several PropertySources and the Consul has higher priority, removing the value, means take from the other property source.

(suggested fix would be to check that the currentIndex is not -1L (but be careful of the "if" later in the code. line 105).

Thanks for your help
Shalom Deitch

**Comment by spencergibb** [comment]
Closing, should be in spring-cloud-consul. Is this a duplicate of https://github.com/spring-cloud/spring-cloud-consul/issues/278?

**Comment by ShalomDeitch1** [comment]
hi,
 Thanks for your quick reply, sorry that I was not as fast as you.
 I reported two issues, this one and https://github.com/spring-cloud/spring-cloud-config/issues/635
 I think that 278 is the same as 635, not the same as this one, 636.
 (I indicated the lines where the issues occur in each report).

Thanks for your help.
Fixing these issues are important to us, as the contents of Consul can change at any time, including having no values, or a new value that should override a value from a different source. 
Shalom Deitch
